### PR TITLE
Fix #211: ListView::itemViewAttributes() accepts Closure as parameter.

### LINKED ISSUE #211: Allow Closure as a valid parameter for `ListView::itemViewAttributes`.
> @samdark @vjik what do you think about allowing `ListView::itemViewAttributes()` to also accept a Closure so the `li` (or other item tag) could be customized according to the specific item?

I agree with @samdark. See as example:
https://github.com/yiisoft/yii-dataview/blob/848da41ead3407cea7a3647e8db66058d9a33708/src/GridView.php#L705

Suggest implement it in separate PR only.

_Originally posted by @vjik in https://github.com/yiisoft/yii-dataview/issues/210#issuecomment-2336597346_

### PR DESCRIPTION
**Pull request by glpzzz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

Refs #211

### CODE REVIEW

File: src/ListView.php [hunk 1]
```diff
@@ -25,7 +25,7 @@ final class ListView extends BaseListView
      */
     private $itemView = null;
 
-    private array $itemViewAttributes = [];
+    private array|Closure $itemViewAttributes = [];
```
Comment (vjik) [inline src/ListView.php:28]: Allow `Closure` to each attribute also, for example:

```php
[
  'class' => static fn() => ...
]
```

**Review by vjik — CHANGES_REQUESTED** [review]

File: src/ListView.php [hunk 2]
```diff
@@ -185,8 +192,18 @@ protected function renderItem(array|object $data, mixed $key, int $index): strin
             $content = (string) call_user_func($this->itemView, $data, $key, $index, $this);
         }
 
+        $itemViewAttributes = is_callable($this->itemViewAttributes)
+            ? (array)call_user_func($this->itemViewAttributes, $data, $key, $index, $this)
+            : $this->itemViewAttributes;
+
+        foreach ($itemViewAttributes as $i => $attribute) {
+            if (is_callable($attribute)) {
+                $itemViewAttributes[$i] = $attribute($data, $key, $index, $this);
```
Comment (samdark) [inline src/ListView.php:201]: This case should be mentioned in phpdoc ↑

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/ListView.php [hunk 3]
```diff
@@ -166,11 +166,25 @@ public function itemViewTag(?string $tag): self
     }
 
     /**
-     * return new instance with the HTML attributes for the container of item view.
+     * Return new instance with the HTML attributes for the container of item view.
+     * If this property is specified as a callback, it must return an array of attributes and have the following
```
Comment (samdark) [inline src/ListView.php:170]:
```suggestion
     * If this property is specified as a function, it must return an array of attributes and have the following
```

or 

```suggestion
     * If this property is specified as a Closure, it must return an array of attributes and have the following
```

File: src/ListView.php [hunk 4]
```diff
@@ -166,11 +166,25 @@ public function itemViewTag(?string $tag): self
...
+     * signature:
      *
-     * @param array $values Attribute values indexed by attribute names.
+     * ```php
+     * function ($data, $key, $index, $widget)
+     * ```
+     *
+     * Also, each attribute value can be a closure too, with the same signature as in:
```
Comment (samdark) [inline src/ListView.php:177]:
```suggestion
     * Also, each attribute value can be a function too, with the same signature as in:
```

File: src/ListView.php [hunk 5]
```diff
@@ -166,11 +166,25 @@ public function itemViewTag(?string $tag): self
...
+     *
+     * ```php
+     * [
+     *     'class' => static fn($data, $key, $index, $widget) => "custom-class-{$data['id']}",
+     * ]
+     * ```
+     * @param array|Closure $values Attribute values indexed by attribute names.
```
Comment (samdark) [inline src/ListView.php:184]: Part of the description above is about this param.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
@vjik I think everything is addressed. Please re-check.

**Review by vjik — APPROVED** [review]

**Comment by vjik** [comment]
@glpzzz Thanks 👍

**Pull request merged**